addon/doxmlparser/examples/doxy2plantuml/main.cpp

```diff
@@ -0,0 +1,222 @@
+
+#include <doxmlintf.h>
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include <fstream>
+#include <map>
+#include <filesystem>
+#include <stack>
+
+
+
+int main(int argc, char **argv) {
+	std::cout << "doxy2plantuml: started" << std::endl;
+
+	std::string cwd;
+	if (argc > 1) {
+		cwd = std::filesystem::path(argv[1]);
+	} else {
+		cwd = std::filesystem::path(argv[0]).parent_path();
+	}
+
+	std::string xml_dir = (cwd + "/xml/");
+
+	IDoxygen *doxygen = createObjectModel();
+	if (!doxygen->readXMLDir(xml_dir.c_str())) {
+		throw std::runtime_error(xml_dir + " not found");
+	}
+
+
+	std::stack<IGraph*> collaboration_graph_stack;
+	std::stack<IGraph*> inheritance_graph_stack;
+
+	std::ofstream outfile("./output.plantuml");
```

Comment:
Wouldn't it be better to specify the resulting file as an argument to the program (with a possible default)?

Reply:
added optional filename argument